### PR TITLE
test/raft: randomize create_test_log() to harden storage tests

### PR DESCRIPTION
**Pull request by emdroid**
Randomize the test log generator in raft_sys_table_storage_test to cover more edge cases: variable length (including empty), randomized first index, random entry types (command/configuration/dummy), random command payloads, random configuration member counts, and monotonically non-decreasing terms with random increments.

create_test_log() now takes optional min_entries/max_entries bounds (default 0..20). Tests that previously relied on a fixed, hand-crafted log with hardcoded indices (truncation, snapshot, replay position handle tests) now call create_test_log(3, 20) and derive the indices and terms they check from the generated entries instead of literal constants, so they keep working with a randomized log.

A failure can be reproduced by rerunning with the printed --random-seed=<N>.

Round-trip tests (store → load → compare) now exercise randomized logs, improving coverage of serialization/deserialization corner cases.

Fixes: SCYLLADB-2429

No backport: This is test update in scope of strong consistency testing (a new feature in master) that does not need backporting.

### CODE REVIEW

File: test/raft/raft_sys_table_storage_test.cc [hunk 1]
```diff
@@ -81,6 +85,83 @@ static std::vector<raft::log_entry_ptr> create_test_log() {
     };
 }
 
+// Create a randomized test log to harden Raft storage tests.
+// Randomizes: length, first index, entry types, command payloads,
+// configuration members, and terms (monotonically non-decreasing).
+// SCYLLADB-2429
+// To reproduce a failure, pass --random-seed=<N> printed by the test runner.
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc]: We need to make sure this seed is always easy to find. IIRC, we always print it in logs, but let's verify that.

We should also make sure that we always create the same log for each seed. I'm not sure if that's obvious.

File: test/raft/raft_sys_table_storage_test.cc [hunk 2]
```diff
@@ -81,6 +85,83 @@ static std::vector<raft::log_entry_ptr> create_test_log() {
     };
 }
 
+// Create a randomized test log to harden Raft storage tests.
+// Randomizes: length, first index, entry types, command payloads,
+// configuration members, and terms (monotonically non-decreasing).
+// SCYLLADB-2429
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc]: nit: I would remove this lonely "SCYLLADB-2429" from the comment.

File: test/raft/raft_sys_table_storage_test.cc [hunk 3]
```diff
@@ -81,6 +85,83 @@ static std::vector<raft::log_entry_ptr> create_test_log() {
...
+// To reproduce a failure, pass --random-seed=<N> printed by the test runner.
+//
+// Guarantees at least one entry of each type (command, configuration, dummy)
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc]: I don't see how
> variable length (including empty)

and

> Guarantees at least one entry of each type (command, configuration, dummy)

can be true at the same time.

I think we shouldn't enforce at least one entry of each type. Logs missing some types or even completely empty logs are also interesting. We should only make sure that logs with at least one entry of each type are probable.

File: test/raft/raft_sys_table_storage_test.cc [hunk 4]
```diff
@@ -6,6 +6,8 @@
  * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.1
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc:6]: > The old deterministic log is preserved as create_fixed_test_log() for
tests that rely on exact hardcoded indices (truncation, snapshot,
replay position handle tests).

Please check if we can easily change these tests to use the randomized log (unless you have already done it).

File: test/raft/raft_sys_table_storage_test.cc [hunk 5]
```diff
@@ -81,6 +85,83 @@ static std::vector<raft::log_entry_ptr> create_test_log() {
...
+// To reproduce a failure, pass --random-seed=<N> printed by the test runner.
+//
+// Guarantees at least one entry of each type (command, configuration, dummy)
+// so every serializer is exercised every run.  The three seed entries are
+// shuffled together with [0..17] additional random-type entries so their
+// positions vary across runs.  Terms are recomputed after the shuffle so
+// they remain monotonically non-decreasing.  server_id values inside
+// configuration entries are built from tests::random so the seed is fully
+// reproducible.
+static std::vector<raft::log_entry_ptr> create_test_log() {
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc]: Printing the generated log could be helpful in debugging. IIUC, we would currently have to rerun the test with the same seed and manually add some logs to see what log is generated for this seed. That sounds annoying.

**Comment by emdroid** [comment]
New version:
* remove the fixed log version completely
* adjust all remaining tests to be able to use the new randomized log
* allow to specify min and max count of log entries, with sensible defaults

[diff](https://github.com/scylladb/scylladb/compare/3789c7efbc5f980dbfa06f2a23a2999a4e03045a..40ca53300c21fb69ffe9fd8f086a963a3db2ffcd)

File: test/raft/raft_sys_table_storage_test.cc (hunk 1 above)
Comment (emdroid) [reply test/raft/raft_sys_table_storage_test.cc]: Yes, verified that the seed is printed in the log.

File: test/raft/raft_sys_table_storage_test.cc (hunk 2 above)
Comment (emdroid) [reply test/raft/raft_sys_table_storage_test.cc]: Removed.

File: test/raft/raft_sys_table_storage_test.cc (hunk 3 above)
Comment (emdroid) [reply test/raft/raft_sys_table_storage_test.cc]: Done, allowing also the empty number of entries. If choosing randomly, we should have probability of empty log: ~4.7%, at least 3 entries: ~86%, at least one entry of each type: ~74%

File: test/raft/raft_sys_table_storage_test.cc (hunk 4 above)
Comment (emdroid) [reply test/raft/raft_sys_table_storage_test.cc:6]: Yes, updated the tests to also use the randomized log. Added a parametrization of the log create function where these test can choose a different minimal number of entries.

File: test/raft/raft_sys_table_storage_test.cc (hunk 5 above)
Comment (emdroid) [reply test/raft/raft_sys_table_storage_test.cc]: Added printing of the generated log entries.

File: test/raft/raft_sys_table_storage_test.cc [hunk 6]
```diff
@@ -56,29 +59,80 @@ using namespace service::strong_consistency;
...
+            .term = raft::term_t(term),
+            .idx  = raft::index_t(first_idx + i),
+            .data = std::move(data_vec[i])}));
+    }
+
+    // Log generated entries to aid reproduction: rerun with same --random-seed=<N>.
+    testlog.info("create_test_log: {} entries, first_idx={}", entries.size(), first_idx);
+    for (const auto& e : entries) {
+        const char* type = std::holds_alternative<raft::command>(e->data)       ? "command"
+                         : std::holds_alternative<raft::configuration>(e->data) ? "configuration"
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc:130]: We could also log the command/configuration details, but I'm not sure if it's worth the effort.

File: test/raft/raft_sys_table_storage_test.cc [hunk 7]
```diff
@@ -56,29 +59,80 @@ using namespace service::strong_consistency;
...
+    size_t count = tests::random::get_int(min_entries, max_entries);
+
+    using data_t = std::variant<raft::command, raft::configuration, raft::log_entry::dummy>;
+    std::vector<data_t> data_vec;
+    data_vec.reserve(count);
+    for (size_t i = 0; i < count; ++i) {
+        switch (tests::random::get_int(0, 2)) {
+        case 0: data_vec.push_back(make_cmd_data());           break;
+        case 1: data_vec.push_back(make_config_data());        break;
+        case 2: data_vec.push_back(raft::log_entry::dummy()); break;
```
Comment (patjed41) [inline test/raft/raft_sys_table_storage_test.cc]: nit: `break`s are misaligned

**Review by patjed41 — APPROVED** [review]
Looks good to me, waiting for @bitpathfinder's approval

File: test/raft/raft_sys_table_storage_test.cc (hunk 7 above)
Comment (emdroid) [reply test/raft/raft_sys_table_storage_test.cc]: fixed

**Review by bitpathfinder — APPROVED** [review]
